codegen-client/src/test/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/ServiceConfigGeneratorTest.kt

```diff
@@ -76,45 +85,100 @@ internal class ServiceConfigGeneratorTest {
+    @ParameterizedTest
+    @ValueSource(strings = ["middleware", "orchestrator"])
+    fun `generate customizations as specified`(smithyRuntimeModeStr: String) {
+        class ServiceCustomizer(private val codegenContext: ClientCodegenContext) :
+            NamedCustomization<ServiceConfig>() {
+            private val runtimeMode = codegenContext.smithyRuntimeMode
+
             override fun section(section: ServiceConfig): Writable {
                 return when (section) {
                     ServiceConfig.ConfigStructAdditionalDocs -> emptySection
-                    ServiceConfig.ConfigStruct -> writable { rust("config_field: u64,") }
+                    ServiceConfig.ConfigStruct -> writable {
+                        if (runtimeMode.defaultToMiddleware) {
+                            rust("config_field: u64,")
+                        }
+                    }
+
                     ServiceConfig.ConfigImpl -> writable {
-                        rust(
-                            """
-                            pub fn config_field(&self) -> u64 {
-                                self.config_field
-                            }
-                            """,
-                        )
+                        if (runtimeMode.defaultToOrchestrator) {
+                            rustTemplate(
+                                """
+                                ##[allow(missing_docs)]
+                                pub fn config_field(&self) -> u64 {
+                                    self.inner.load::<#{T}>().map(|u| u.0).unwrap()
+                                }
+                                """,
+                                "T" to configParamNewtype(
+                                    "config_field".toPascalCase(), RuntimeType.U64.toSymbol(),
+                                    codegenContext.runtimeConfig,
+                                ),
+                            )
+                        } else {
+                            rust(
+                                """
+                                ##[allow(missing_docs)]
+                                pub fn config_field(&self) -> u64 {
+                                    self.config_field
+                                }
+                                """,
+                            )
+                        }
                     }
+
                     ServiceConfig.BuilderStruct -> writable { rust("config_field: Option<u64>") }
                     ServiceConfig.BuilderImpl -> emptySection
                     ServiceConfig.BuilderBuild -> writable {
-                        rust("config_field: self.config_field.unwrap_or_default(),")
+                        if (runtimeMode.defaultToOrchestrator) {
+                            rustTemplate(
+                                """
+                                layer.store_or_unset(self.config_field.map(#{T}));
+                                """,
```

Comment:
If it's just a single line and doesn't contain any double quotes, you don't need triple quotes:
```suggestion
                                "layer.store_or_unset(self.config_field.map(#{T}));",
```
Saves a few lines.